server/liveblog/themes/themes.py

```diff
@@ -174,6 +174,28 @@ def save_or_update_theme(self, theme, files=[], force_update=False):
                         theme['screenshot_url'] = superdesk.upload.url_for_media(file_id)
         previous_theme = self.find_one(req=None, name=theme.get('name'))
         if previous_theme:
+            blogs_service = get_resource_service('blogs')
+            # retrieve the blog that use the previous theme
+            blogs = blogs_service.get(req=None, lookup={'blog_preferences.theme': previous_theme['name']})
+            # retrieve the default settings of the current theme
+            default_theme_settings = get_resource_service('themes').get_default_settings(theme)
+            # retrieve the default settings of the previous theme
+            default_prev_theme_settings = get_resource_service('themes').get_default_settings(previous_theme)
+
+            for blog in blogs:
+                for key, value in blog['theme_settings'].items():
+                    # if the values of theme setting from blog level are the same as the settings
+                    # from previous theme update the settings
+                    if value == default_prev_theme_settings[key]:
+                        blogs_service.system_update(ObjectId(blog['_id']),
```

Comment:
It's better to not save the blog after each settings but at the end, before doing the same for the next blog.